tools/db/phone-users/src/PhoneUsers/Lib.hs

```diff
@@ -0,0 +1,172 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2024 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module PhoneUsers.Lib where
+
+import Cassandra as C
+import Cassandra.Settings as C
+import Data.Conduit
+import qualified Data.Conduit.Combinators as Conduit
+import Data.Id (TeamId, UserId)
+import Data.Time
+import qualified Database.CQL.Protocol as CQL
+import Imports
+import Options.Applicative
+import PhoneUsers.Types
+import qualified System.IO as SIO
+import qualified System.Logger as Log
+import Wire.API.Team.Feature (FeatureStatus (FeatureStatusDisabled, FeatureStatusEnabled))
+import Wire.API.User (AccountStatus (Active))
+
+lookupClientsLastActiveTimestamps :: ClientState -> UserId -> IO [Maybe UTCTime]
+lookupClientsLastActiveTimestamps client u = do
+  runClient client $ runIdentity <$$> retry x1 (query selectClients (params LocalQuorum (Identity u)))
```

Comment:
I think we shouldn't use quorum reads here, they are more expensive and its not terribly important that this data is super up to date.